File: api/app/routers/ai.py
import select

from api.app.db.models.users import User
from fastapi import APIRouter, Depends
from pydantic import BaseModel
from sqlalchemy.ext.asyncio import AsyncSession
from app.db.database import get_db
from app.services.ai_service import get_action_type

router = APIRouter(prefix="/ai", tags=["ai"])


class MessageRequest(BaseModel):
    text: str
    discord_id: str


class MessageResponse(BaseModel):
    reply: str


@router.post("/message", response_model=MessageResponse)
async def handle_message(request: MessageRequest, db: AsyncSession = Depends(get_db)):

    result = await db.execute(select(User).where(User.discord_id == request.discord_id))
    user = result.scalar_one_or_none()

    if not user or not user.is_ai:
        return {"reply": "Nie masz dostępu do AI."}

    result = await get_action_type(request.text, request.discord_id, db)
    return result

File: api/src/routers/ai.ts
import { Router, Request, Response } from "express";
import { prisma } from "../db/prisma";
import { getActionType } from "../services/aiService";
import { MESSAGES } from "../messages";

const router = Router();

router.post("/message", async (req: Request, res: Response) => {
  const { text, discord_id } = req.body as { text: string; discord_id: string };

  const user = await prisma.user.findUnique({ where: { discordId: discord_id } });
  if (!user || !user.isAi) {
    return res.json({ reply: MESSAGES.NO_AI_ACCESS });
  }

  const result = await getActionType(text, discord_id);
  return res.json(result);
});

export default router;
